source/javascripts/app/_search.js

```diff
@@ -29,22 +29,36 @@
     });
   }
 
+  var wait = (
+      function(){
+        var timeout_handle = 0;
+        return function(executing_function, wait_time){
+          clearTimeout (timeout_handle);
+          timeout_handle = setTimeout(executing_function, wait_time);
+        };
+      })();
+
   function bind() {
     content = $('.content');
     searchResults = $('.search-results');
 
-    $('#input-search').on('keyup', search);
+    var search_timeout = parseFloat($('#input-search').attr('search_timeout') || 0);
+    $('#input-search').on('keyup',function(e) {
+        wait(function(){
+          search(e,$('#input-search')[0]);
+        }, search_timeout );
+    });
   }
 
-  function search(event) {
+  function search(event,search_input) {
```

Comment:
Considering there will only ever be one `search_input`, maybe it makes sense to just say `var search_input = $('#input-search')` in this function, instead of having another argument? Or call `search.apply`, although I guess that's kinda a hacky solution too. Well, your call, if you have a good argument for having this passed into the function, we can stick with this solution too.